### PR TITLE
[opentelemetry-operator] Add targets to update and check CRDs

### LINKED ISSUE #298: Add make target to auto-update operator CRDs
Would be good if we could add a make target in this repo that would automatically update the CRDs. Then we could run a test in the CI similar to Check Examples that would verify things. 

__Originally posted by @TylerHelmuth in https://github.com/open-telemetry/opentelemetry-helm-charts/pull/297#issuecomment-1200170334__

### PR DESCRIPTION
**Pull request by TylerHelmuth**
Adds new make targets to update and check the operator CRDs.  

Closes #298

### CODE REVIEW

**Review by Allex1 — APPROVED** [review]

**Review by dmitryax — COMMENTED** [review]
Nice! Thanks @TylerHelmuth. Can you please all update this doc accordingly https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-operator/release/release-checklist.md

**Comment by TylerHelmuth** [comment]
> Nice! Thanks @TylerHelmuth. Can you please all update this doc accordingly https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-operator/release/release-checklist.

Thanks for bringing this to my attention. I actually think I need to review the whole release folder, it looks like it also automatically updates the CRDs.

**Comment by TylerHelmuth** [comment]
@Allex1 have you been using this release folder for the operator bumps?

**Comment by Allex1** [comment]
> @Allex1 have you been using this release folder for the operator bumps?

@TylerHelmuth  yes, I manually updated crds in `./charts/opentelemetry-operator/crds/` when bumping to 0.56.0 https://github.com/open-telemetry/opentelemetry-helm-charts/commit/628c218a46d619ba57fc86008ebd95fc02fe5074

**Comment by TylerHelmuth** [comment]
@Allex1 take a look at https://github.com/open-telemetry/opentelemetry-helm-charts/pull/324 when you get a chance, I propose we remove the code in the release folder

**Comment by TylerHelmuth** [comment]
@dmitryax @Allex1 how do you feel about a target like this:

```
.PHONY: update-crds-2
update-crds-2:
	APP_VERSION=$$(cat ./charts/opentelemetry-operator/Chart.yaml | sed -nr 's/appVersion: ([0-9]+\.[0-9]+\.[0-9]+)/\1/p') ; \
	git clone -b v$${APP_VERSION} --single-branch https://github.com/open-telemetry/opentelemetry-operator.git ./opentelemetry-operator
	cp -a ./opentelemetry-operator/config/crd/bases/. ./charts/opentelemetry-operator/crds
	rm -rf ./opentelemetry-operator
```

This option gets the entire CRD folder, so if a 3rd CRD is added we don't have to update our makefile.  It does require a rename of those files tho, bc currently our filenames don't match the operator repos

**Comment by dmitryax** [comment]
We might want to be aware of any new CDRs added to the operator. Not sure if we want to blindly fetch everything...

**Comment by TylerHelmuth** [comment]
> Not sure if we want to blindly fetch everything...

Ya thats how I felt too.  The old release code got the CRDs from the operator's manifest at `https://github.com/open-telemetry/opentelemetry-operator/releases/latest/download/opentelemetry-operator.yaml`.  Or we could stick with this PR's current implementation.

File: Makefile [hunk 1]
```diff
@@ -39,3 +39,23 @@ check-examples:
 		done; \
 	done
 
+.PHONY: update-crds
+update-crds:
```
Comment (dmitryax) [inline Makefile:43]: Let's call them `update-operator-crds` and `check-operator-crds`?

Comment (TylerHelmuth) [reply Makefile]: Done

**Comment by Allex1** [comment]
> We might want to be aware of any new CDRs added to the operator. Not sure if we want to blindly fetch everything...

I've mixed feelings as well: as I commented in https://github.com/open-telemetry/opentelemetry-helm-charts/issues/310 it's not like we make any use of the actual crds in this helm chart (other than deploying them in the cluster). So it might be useful for some users to get any new crds as soon as they become available in order to start using them. On the other hand it seems like a trivial change to add any eventual extra crds ... so I'm fine either way.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**